### ISSUE TITLE
Why "greater than or equal to" rather than "greater than"?

### ISSUE THREAD

**Issue by elemoine** [opengisch/pum#72]
I am opening this issue for a question.

Pum ugrade applies all the deltas that haven't been applied yet and whose versions are greater than or equal to the current db version. Why "greater than or equal to" rather than just "greater than"?

I find this strategy a bit problematic, because it makes it hard for an application user to choose a baseline.

Let's say I am installing an application, and the version I am installing includes delta files with versions 0.0.1, 0.0.2 and 0.0.3. What should be my baseline? If I choose 0.0.3 for the baseline then the next `pum upgrade` will apply the 0.0.3 delta, which I don't want, because the database already has the structure brought by the 0.0.3 delta. So I should probably choose 0.0.4 for the baseline. Is that correct?

Thanks.

**Comment by 3nids** [comment]
I don't see any valid and agree with you.
https://github.com/opengisch/pum/pull/73

**Comment by marioba** [comment]
I think the initial idea was to consider a "delta_x" as a _patch_ for the "version_x" of the database. I agree it's not very intuitive, especially with baselines.

**Comment by elemoine** [comment]
One thing: the current approach supports having multiple delta files with the same version. Say you have delta_0.0.1_a.sql. You run pum upgrade, and the current db version changes 0.0.1. Then you add delta_0.0.1_b.sql. You run pum upgrade again and delta_0.0.1_b.sql will be applied. This is particularly convenient when you use a development version of an application, and want to apply all the new deltas after an application update.

**Comment by elemoine** [comment]
I think what I would like to do is just change the way the baseline works. This is what I'd like:

When I set the baseline to, say, 1.2.3, I want pum upgrade to only apply deltas that are > 1.2.3. But after I've applied new deltas, say 1.2.4 deltas, I want pum upgrade to apply deltas that are >= 1.2.4. Bottom line: I'd like to make the baseline to be treated as a special case, because I think it's a special case. Because when I install version x.y.z of an application I want to set the baseline to x.y.z, and I don't want pum upgrade to apply x.y.z deltas.

See https://github.com/opengisch/pum/pull/77 for a WIP PR implementing this.

**Comment by elemoine** [comment]
I created #77 for discussion only. I am really not sure about it. I'd like to think more about the whole thing, and make a few tests before being settled on this.

**Comment by elemoine** [comment]
After more thinking we've decided to use the "consider a delta_x as a patch for the version_x of the database" paradigm that @marioba mentioned in https://github.com/opengisch/pum/issues/72#issuecomment-587037277.

So when the latest release of our application is, say, 0.5.0, we add deltas to our code base with version 0.5.0. They are the deltas that users will need to apply to migrate from 0.5.0.

We'll see how this works for us, and report back here. And if it does work we could document that method somewhere in the Pum documentation.

For your information, we've decided to advertize the use of Pum for the migrations of [Procrastinate](https://github.com/peopledoc/procrastinate).
